server/src/main/java/org/elasticsearch/action/support/ChannelActionListener.java

```diff
@@ -28,6 +28,7 @@ public ChannelActionListener(TransportChannel channel) {
 
     @Override
     public void onResponse(Response response) {
+        response.incRef(); // acquire reference that will be released by channel.sendResponse below
```

Comment:
On reflection I think it'd be nicer to push this down a level or two and make `TransportChannel#sendResponse` refcount-neutral. In particular it's kinda weird that `OutboundHandler#sendRequest` is already refcount-neutral whilst `OutboundHandler#sendResponse` is not.

Reply:
Yea it sucks but could we push that out to another PR? That's practically a rather big change it seems, just tried it. We have a number of spots outside of this listener where we call the channel.sendResponse() directly and I'd have to adjust all of those. That goes beyond the scope of what I'm trying to fix here. Note that this at least gets us closer to fixing the OutBoundHandler :) ok with you?

Reply:
Yeah ok makes sense.